### ISSUE TITLE
bug: clean_pykeops does not clean up the kernels in custom bin_folder

### ISSUE THREAD

**Issue by mihai-spire** [getkeops/keops#95]
**Problem**:

Calling `cleanup_keops` after setting a custom `bin_dir` does _not_ remove the kernels present in `bin_dir`.
Instead, `cleanup_keops` always tries to remove stuff under the default kernel bin path (under `$home`).
If nothing's there then I get a `FileNotFoundError`.

NB: same thing happens if I build some kernels in the custom `bin_dir` before cleaning.

**How to reproduce**:

```
>>> import pykeops
>>> pykeops.verbose = True
>>> pykeops.set_bin_folder('/scratch/custom_bin_folder')
>>> pykeops.clean_pykeops()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "user/.conda/envs/DEV/lib/python3.7/site-packages/pykeops/common/utils.py", line 133, in clean_pykeops
    for f in os.listdir(path):
FileNotFoundError: [Errno 2] No such file or directory: 'user/.cache/pykeops-1.4.1-cpython-37/'
```

**Expected behavior**:
All kernels under `/scratch/custom_bin_folder` should be removed.

`pykeops` version: 1.4.1 + [`724bcf2`](https://github.com/getkeops/keops/commit/724bcf2fe19a0f122ab747ba65494054e42549d3) + [2c96e93](https://github.com/getkeops/keops/commit/2c96e93e06c3d0de6ed46baf03dafb56917ef17d)
OS: Linux

**Comment by mihai-spire** [comment]
also, i've noticed that a 2nd call to `set_bin_folder` with a different `bin_folder` path has no effect, i.e. `keops` does not re-compile the kernels in the 'new' bin_folder - presumably, it still uses the kernels in the 'first' `bin_folder`. you can reproduce this with:

```
>>> import torch, pykeops
>>> torch.cuda.is_available()
True
>>> pykeops.set_bin_folder('/scratch/bin-folder1')
>>> pykeops.test_torch_bindings()
Compiling libKeOpstorch2a3cde5827 in /scratch/bin-folder1/build-libKeOpstorch2a3cde5827:
       formula: Sum_Reduction(SqNorm2(x - y),1)
       aliases: x = Vi(0,3); y = Vj(1,3); 
       dtype  : float32
... Generated /scratch/bin-folder1/build-libKeOpstorch2a3cde5827/CMakeFiles/keopslibKeOpstorch2a3cde5
827.dir/keops/core/./keopslibKeOpstorch2a3cde5827_generated_link_autodiff.cu.o successfully.
Done.
pyKeOps with torch bindings is working!
>>> pykeops.set_bin_folder('/scratch/bin-folder2')
>>> pykeops.test_torch_bindings()
pyKeOps with torch bindings is working!
```

`/scratch/bin-folder2` is created but left empty
```
$ls -l /scratch/bin-folder2
total 0
```

**Comment by mihai-spire** [comment]
@bcharlier see also https://github.com/getkeops/keops/issues/85#issuecomment-693261429 thanks!

**Comment by bcharlier** [comment]
Hi @mihai-spire ,

the last commit on master should do the trick:

```python
>>> import pykeops
>>> pykeops.clean_pykeops()
Cleaning /home/bcharlier/projets/keops/keops/pykeops/build/...
    - /home/bcharlier/projets/keops/keops/pykeops/build/libKeOpsnumpy0a5fc8b901.cpython-38-x86_64-linux-gnu.so has been removed.
    - /home/bcharlier/projets/keops/keops/pykeops/build/libKeOpsnumpy0a5fc8b901.so has been removed.
    - /home/bcharlier/projets/keops/keops/pykeops/build/.gitkeep has been removed.
    - /home/bcharlier/projets/keops/keops/pykeops/build/keops_hash.log has been removed.
>>> pykeops.test_numpy_bindings()
Compiling libKeOpsnumpy0a5fc8b901 in /home/bcharlier/projets/keops/keops/pykeops/build:
       formula: Sum_Reduction(SqNorm2(x - y),1)
       aliases: x = Vi(0,3); y = Vj(1,3); 
       dtype  : float64
 Generated /home/bcharlier/projets/keops/keops/pykeops/build/build-libKeOpsnumpy0a5fc8b901/CMakeFiles/keopslibKeOpsnumpy0a5fc8b901.dir/keops/core/./keopslibKeOpsnumpy0a5fc8b901_generated_link_autodiff.cu.o successfully.
Done.
pyKeOps with numpy bindings is working!
>>> pykeops.set_bin_folder("~/test_keops_bin_folder")
/home/bcharlier/projets/keops/keops/pykeops/common/set_path.py:45: UserWarning: [pyKeOps]: set_bin_folder() has been invoked while some pykeops modules have already been imported. To avoid this warning, change pykeops.config.bin_folder just after importing pykeops, that is, before any computations.
  warnings.warn("[pyKeOps]: set_bin_folder() has been invoked while some pykeops modules have already "
>>> pykeops.test_numpy_bindings()
Compiling libKeOpsnumpy1f54fb16f7 in /home/bcharlier/test_keops_bin_folder:
       formula: Sum_Reduction(SqNorm2(x - y),1)
       aliases: x = Vi(0,3); y = Vj(1,3); 
       dtype  : float64
 Generated /home/bcharlier/test_keops_bin_folder/build-libKeOpsnumpy1f54fb16f7/CMakeFiles/keopslibKeOpsnumpy1f54fb16f7.dir/keops/core/./keopslibKeOpsnumpy1f54fb16f7_generated_link_autodiff.cu.o successfully.
Done.
pyKeOps with numpy bindings is working!
>>> pykeops.clean_pykeops()
Cleaning /home/bcharlier/test_keops_bin_folder/...
    - /home/bcharlier/test_keops_bin_folder/libKeOpsnumpy1f54fb16f7.cpython-38-x86_64-linux-gnu.so has been removed.
    - /home/bcharlier/test_keops_bin_folder/libKeOpsnumpy1f54fb16f7.so has been removed.
    - /home/bcharlier/test_keops_bin_folder/keops_hash.log has been removed.
```

Note that a warning is thrown when changing the `bin_folder` if a formula has already been imported. The reason is : I don't know how to unload modules from the old `bin_folder`. It works, but it is not very safe -- from my point of view --  to handle modules compiled from different folders...

**Comment by bcharlier** [comment]
hi @mihai-spire ,

it should be fixed with the new release 1af2124. Feel free to reopen if needed.
